Documentation/gettingstarted/local-redirect-policy.rst

```diff
@@ -301,5 +309,157 @@ configuring the CiliumLocalRedirectPolicy.
 Local Redirect Policy updates are currently not supported. If there are any
 changes to be made, delete the existing policy, and re-create a new one.
 
+Use Cases
+=========
+Local Redirect Policy allows Cilium to support the following use cases:
+
+Node-local DNS cache
+--------------------
+`DNS node-cache <https://github.com/kubernetes/dns>`_ listens on a static IP to intercept
+traffic from application pods to the cluster's DNS service VIP by default, which will be
+bypassed when Cilium is handling service resolution at or before the veth interface of the
+application pod. To enable the DNS node-cache in a Cilium cluster, the following example
+steers traffic to a local DNS node-cache which runs as a normal pod.
+
+* Deploy DNS node-cache in pod namespace.
+
+  .. tabs::
+
+    .. group-tab:: Quick Deployment
+
+        Deploy DNS node-cache.
+
+        .. note::
+
+           The quick deployment example yaml deploys node-local-dns in ``kube-system`` namespace,
+           this does not work for GKE clusters.
```

Comment:
I wonder, does this work if you deploy in a different namespace? Just wondering whether we're trying to tell people there's no way to use this on GKE or whether they just need to do things a little differently.

Reply:
Umm... The node-local dns pods need to be able to connect to the kube dns pods so you would have to deploy both sets of pods in the same namespace. @Weil0ng can confirm this.

Reply:
any pod should be able to talk to kube-dns assuming there isn't a network policy to prevent it. Is there something special or different about node-local DNS here?

Reply:
Ah! Yes, that makes sense. :)

Reply:
ah I don't think so, I never tried in non-kube-system ns thought, but I don't see it cannot work? will try it out now and reply here

Reply:
In order to deploy the pods in some other namespace, I had to comment out this field - `priorityClassName: system-node-critical`. Let's document this.

> Error creating: pods "node-local-dns-" is forbidden: pods with system-node-critical priorityClass is not permitted in cilium namespace

Reply:
It turns out it won't work on GKE because the upstream dns service created by node-local-dns has to be in the same namespace as kube-dns, which is kube-system, which GKE doesn't allow you to deploy random stuff in it.

Reply:
I dropped mention of GKE since this is not how oss nodelocaldns is supposed to work on GKE, which is not caused by/specific to Cilium.

Reply:
Do you see the same error for `priorityClass` in non-GKE clusters? @aditighag

Reply:
I deployed the pod in the kube-system namespace on my non-GKE cluster. I would imagine the constraint to be specific to GKE.